### PR TITLE
Update documentation for queue.rerunTask

### LINKED ISSUE #4587: Update rerunTask docs regarding completed, unscheduled tasks
https://bugzilla.mozilla.org/show_bug.cgi?id=1458896

### PR DESCRIPTION
**Pull request by djmitche**
..and add some tests to verify the new docs are accurate.

Github Bug/Issue: Fixes #4587

### CODE REVIEW

**Comment by djmitche** [comment]
Red x's are
```
error An unexpected error occurred: "https://registry.yarnpkg.com/@slack/logger/-/logger-2.0.0.tgz: Request failed \"500 Internal Server Error\"".
```
we should probably figure out a way to use a yarn cache in our CI..

File: services/queue/test/reruntask_test.js [hunk 1]
```diff
@@ -35,32 +35,64 @@ helper.secrets.mockSuite(testing.suiteName(), ['aws'], function(mock, skipping)
...
     });
     helper.assertPulseMessage('task-running');
+  };
+
+  test('rerun a scheduled task', async () => {
+    // create a self-dependent task, so it is in state 'scheduled'
+    await createTask({ ...taskDef, dependencies: [taskId] });
+
+    debug('### Requesting task rerun');
+    helper.scopes('queue:rerun-task:my-scheduler-extended-extended/dSlITZ4yQgmvxxAi4A8fHQ/*');
```
Comment (petemoore) [inline services/queue/test/reruntask_test.js:68]: How was this task originally created?
Do we need to account for it in any development docs?

**Review by petemoore — APPROVED** [review]

File: services/queue/test/reruntask_test.js (hunk 1 above)
Comment (djmitche) [reply services/queue/test/reruntask_test.js:68]: It's created on line 65 :)

Comment (djmitche) [reply services/queue/test/reruntask_test.js:68]: Ah, the `dslITZ..` slugId is the taskGroupId, which is hard-coded a bit further up.  None of this stuff runs against the production queue -- it's running against the queue under test.

**Pull request merged**